ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/LgrDB.hs

```diff
@@ -81,23 +86,22 @@ import qualified Ouroboros.Consensus.Storage.ChainDB.Impl.BlockCache as BlockCac
 import           Ouroboros.Consensus.Storage.Common
 import           Ouroboros.Consensus.Storage.ImmutableDB (ImmutableDB)
 import qualified Ouroboros.Consensus.Storage.ImmutableDB as ImmutableDB
-import           Ouroboros.Consensus.Storage.LedgerDB (LedgerDB')
+import           Ouroboros.Consensus.Storage.LedgerDB (LedgerDB',
+                     configLedgerDb, ledgerDbCfgSecParam)
 import qualified Ouroboros.Consensus.Storage.LedgerDB as LedgerDB
 import           Ouroboros.Consensus.Storage.LedgerDB.BackingStore
-import           Ouroboros.Consensus.Storage.LedgerDB.DbChangelog hiding (flush)
+import           Ouroboros.Consensus.Storage.LedgerDB.DbChangelog hiding
+                     (splitForFlushing)
 import           Ouroboros.Consensus.Storage.LedgerDB.ReadsKeySets
-                     (PointNotFound, readKeySets)
-import qualified Ouroboros.Consensus.Storage.LedgerDB.ReadsKeySets as ReadsKeySets
 import           Ouroboros.Consensus.Storage.LedgerDB.Stream
 import           Ouroboros.Consensus.Storage.Serialisation
+import           Ouroboros.Consensus.Util
 import           Ouroboros.Consensus.Util.Args
 import           Ouroboros.Consensus.Util.IOLike
-import qualified Ouroboros.Consensus.Util.MonadSTM.RAWLock as Lock
 import           Ouroboros.Consensus.Util.ResourceRegistry
 import           System.FS.API (SomeHasFS (..), createDirectoryIfMissing)
 import           System.FS.API.Types (FsError, mkFsPath)
 
-
 -- | Thin wrapper around the ledger database
 data LgrDB m blk = LgrDB {
       varDB           :: !(StrictTVar m (LedgerDB' blk))
```

Comment:
Tangent: `RAWLock`s are intendented to keep some in-memory state in sync with some on-disk state (as the documentation of `RAWLock` describes it. Maybe it's not a very weird idea that this `varDB` could be a `RAWLock`?, since we're trying to keep the changelog in sync with the backing store? This is just a faint idea I had, so maybe it doesn't make too much sense.

Reply:
Maybe :/